test/unit/math/rev/prob/neg_binomial_2_log_test.cpp

```diff
@@ -0,0 +1,564 @@
+  std::vector<double> eta_to_test = {-2, 0, 0.536844, 1.26845, 11};
+
+  auto nb2_log_for_test = [](int n, const std::complex<double>& eta,
+                             const std::complex<double>& phi) {
+    // Using first-order Taylor expansion of lgamma(a + b*i) around b = 0
+    // Which happens to work nice in this case, as b is always 0 or the very
+    // small complex step
+    auto lgamma_c_approx = [](const std::complex<double>& x) {
+      return std::complex<double>(lgamma(x.real()),
+                                  x.imag() * boost::math::digamma(x.real()));
+    };
+
+    const double n_(n);
+    return lgamma_c_approx(n_ + phi) - lgamma(n + 1) - lgamma_c_approx(phi)
+           + phi * (log(phi) - log(exp(eta) + phi)) - n_ * log(exp(eta) + phi)
+           + n_ * eta;
+  };
+
+  double phi_cutoff = 1e10;
+  for (double eta_dbl : eta_to_test) {
+    for (int n : n_to_test) {
+      for (double phi_dbl = 1.5; phi_dbl < 1e22; phi_dbl *= 10) {
+        var eta(eta_dbl);
+        var phi(phi_dbl);
+        var val = neg_binomial_2_log_lpmf(n, eta, phi);
+
+        std::vector<var> x;
+        x.push_back(eta);
+        x.push_back(phi);
+
+        std::vector<double> gradients;
+        val.grad(x, gradients);
+
+        EXPECT_TRUE(value_of(val) < 0)
+            << "for n = " << n << ", eta = " << eta_dbl
+            << ", phi = " << phi_dbl;
+
+        for (int i = 0; i < 2; ++i) {
+          EXPECT_FALSE(is_nan(gradients[i]));
+        }
+
+        auto nb2_log_eta
+            = [n, phi_dbl, nb2_log_for_test](const std::complex<double>& eta) {
+                return nb2_log_for_test(n, eta, phi_dbl);
+              };
+        auto nb2_log_phi
+            = [n, eta_dbl, nb2_log_for_test](const std::complex<double>& phi) {
+                return nb2_log_for_test(n, eta_dbl, phi);
+              };
+        double complex_step_deta
+            = complex_step_derivative(nb2_log_eta, eta_dbl);
+        double complex_step_dphi
+            = complex_step_derivative(nb2_log_phi, phi_dbl);
+
+        std::ostringstream message;
+        message << ", n = " << n << ", eta = " << eta_dbl
+                << ", phi = " << phi_dbl;
+
+        double tolerance_phi;
+        double tolerance_eta;
+        if (phi < phi_cutoff || n < 100000) {
+          tolerance_phi = std::max(1e-10, fabs(gradients[1]) * 1e-8);
+        } else {
+          tolerance_phi = std::max(1e-8, fabs(gradients[1]) * 1e-5);
+        }
+
+        if (phi < phi_cutoff) {
+          tolerance_eta = std::max(1e-10, fabs(gradients[0]) * 1e-8);
+        } else {
+          tolerance_eta = std::max(1e-8, fabs(gradients[0]) * 1e-5);
+        }
+
+        EXPECT_NEAR(gradients[0], complex_step_deta, tolerance_eta)
+            << "grad_mu" << message.str();
+
+        EXPECT_NEAR(gradients[1], complex_step_dphi, tolerance_phi)
+            << "grad_phi" << message.str();
+      }
+    }
+  }
+}
+
+TEST(mathMixScalFun, neg_binomial_2_log_lpmf_derivatives) {
```

Comment:
Oops, I shoulda caught this. This should be in mix, but given how long the tests have taken merge this anyway and we can just move this to mix with the next pull.